facefusion/ffmpeg.py

```diff
@@ -24,20 +25,21 @@ def open_ffmpeg(args : List[str]) -> subprocess.Popen[bytes]:
 	return subprocess.Popen(commands, stdin = subprocess.PIPE)
 
 
-def extract_frames(target_path : str, fps : float) -> bool:
+def extract_frames(target_path : str, video_resolution : str, video_fps : float) -> bool:
+	video_width, video_height = unpack_resolution(video_resolution)
 	temp_frame_compression = round(31 - (facefusion.globals.temp_frame_quality * 0.31))
 	trim_frame_start = facefusion.globals.trim_frame_start
 	trim_frame_end = facefusion.globals.trim_frame_end
 	temp_frames_pattern = get_temp_frames_pattern(target_path, '%04d')
 	commands = [ '-hwaccel', 'auto', '-i', target_path, '-q:v', str(temp_frame_compression), '-pix_fmt', 'rgb24' ]
 	if trim_frame_start is not None and trim_frame_end is not None:
-		commands.extend([ '-vf', 'trim=start_frame=' + str(trim_frame_start) + ':end_frame=' + str(trim_frame_end) + ',fps=' + str(fps) ])
+		commands.extend([ '-vf', 'trim=start_frame=' + str(trim_frame_start) + ':end_frame=' + str(trim_frame_end) + ',scale=' + str(video_width) + ':' + str(video_height) + ',fps=' + str(video_fps) ])
 	elif trim_frame_start is not None:
-		commands.extend([ '-vf', 'trim=start_frame=' + str(trim_frame_start) + ',fps=' + str(fps) ])
+		commands.extend([ '-vf', 'trim=start_frame=' + str(trim_frame_start) + ',scale=' + str(video_width) + ':' + str(video_height) + ',fps=' + str(video_fps) ])
 	elif trim_frame_end is not None:
-		commands.extend([ '-vf', 'trim=end_frame=' + str(trim_frame_end) + ',fps=' + str(fps) ])
+		commands.extend([ '-vf', 'trim=end_frame=' + str(trim_frame_end) + ',scale=' + str(video_width) + ':' + str(video_height) + ',fps=' + str(video_fps) ])
 	else:
-		commands.extend([ '-vf', 'fps=' + str(fps) ])
+		commands.extend([ '-vf', 'scale=' + str(video_width) + ':' + str(video_height) + ',fps=' + str(video_fps) ])
```

Comment:
change name from `fps` to `video_fps`
and add validation to` fps`, `video_width` and `video_height`

Reply:
will fix that in post on next branch